examples/dynamic_watcher.rs

```diff
@@ -22,18 +22,18 @@ async fn main() -> anyhow::Result<()> {
     // Turn them into a GVK
     let gvk = GroupVersionKind::gvk(&group, &version, &kind);
     // Use API discovery to identify more information about the type (like its plural)
-    let (ar, caps) = discovery::pinned_kind(&client, &gvk).await?;
+    let ar = discovery::pinned_kind(&client, &gvk).await?;
 
     // Use the full resource info to create an Api with the ApiResource as its DynamicType
     let api = Api::<DynamicObject>::all_with(client, &ar);
 
     // Fully compatible with kube-runtime
     let mut items = watcher(api, ListParams::default()).applied_objects().boxed();
     while let Some(p) = items.try_next().await? {
-        if caps.scope == Scope::Cluster {
-            info!("saw {}", p.name_any());
-        } else {
+        if ar.namespaced {
             info!("saw {} in {}", p.name_any(), p.namespace().unwrap());
+        } else {
+            info!("saw {}", p.name_any());
         }
```

Comment:
```suggestion
        if let Some(ns) = p.namespace() {
            info!("saw {} in {}", p.name_any(), ns);
        } else {
            info!("saw {}", p.name_any());
        }
```

Reply:
Minor nitpick but I don't think we should encourage avoidable unwraps in example code.